frontend/utils/distribution/monthly-event-parser.js

```diff
@@ -54,11 +52,49 @@ function subtractDistributions (paymentsA: Distribution, paymentsB: Distribution
 
   return addDistributions(paymentsA, paymentsB)
 }
+// Create a helper function for calculating each cycle's payment distribution:
+function paymentsDistribution (groupMembers, payments, mincome, adjusted, minimizeTxns) {
+  const groupIncomes = groupMembers.map((user) => {
+    return {
+      name: user.name,
+      amount: mincome + user.haveNeed
+    }
+  })
+
+  let distribution = incomeDistribution(groupIncomes, mincome)
+
+  distribution = minimizeTxns ? minimizeTotalPaymentsCount(distribution, groupMembers) : distribution
+  distribution = distribution.map((payment) => {
+    payment.total = payment.amount
+    return payment
+  })
+
+  distribution = subtractDistributions(distribution, payments)
+
+  const havers = groupMembers.filter(m => m.haveNeed > 0)
+  let ratio = 1
+  for (const haver of havers) {
+    const paymentsFrom = payments.filter(p => p.from === haver.name)
+    const totalPaidFrom = paymentsFrom.reduce((a, p) => a + p.amount, 0)
+    const distributionFrom = distribution.filter(p => p.from === haver.name)
+    const totalDistFrom = distributionFrom.reduce((a, p) => a + p.amount, 0)
+    const scalar = (totalDistFrom) / (haver.haveNeed - totalPaidFrom)
+    ratio = Math.max(ratio, scalar)
+  }
+
+  distribution = distribution.map(p => {
+    p.amount /= ratio
+    p.total /= ratio
+    return p
+  })
```

Comment:
Is this the clamping code? If so, then is it possible to create a single clamping function to pass the distribution to and have it return a clamped version? Or something to that effect? (Have a function with the word "clamped" / "clamp" in it will also make it clear what the code is doing).

Reply:
Yes and yes. I'll get right on it.

Also: is it ok if we say "scaling"? Clamping A between B and C refers to following operation (in JS):
`Math.max(Math.min(A, C), B)`

Where as this is actually scalar multiplication.